### ISSUE TITLE
`both` -> `conditional` migration issue - different coercion behaviour

### ISSUE THREAD

**Issue by cichli** [Prismatic/schema#291]
Strongly related to #269; feel free to close if this is effectively a duplicate.

We've hit an issue with coercion while migrating our codebase from `both` to `conditional`. Previously, we could do the following (where `schema.core` and `schema.coerce` are aliased to `sc` and `sco` respectively):

```clojure
user> (let [schema (sc/both sc/Int (sc/pred pos?))
            coercer (sco/coercer schema sco/string-coercion-matcher)]
        (coercer "1"))
1
```

I haven't figured out a way to accomplish the same thing with `conditional`:

```clojure
user> (let [schema (sc/conditional pos? sc/Int)
            coercer (sco/coercer schema sco/string-coercion-matcher)]
        (coercer "1"))
#schema.utils.ErrorContainer{:error (throws? (pos? "1"))}
```

Would the suggested post-condition variant of `conditional` suggested in #269 resolve this issue?

**Comment by w01fe** [comment]
Thanks for the report -- I think https://github.com/Prismatic/schema/pull/292 should solve this (see the new coercion test) -- look good to you?

**Comment by cichli** [comment]
Thanks for the quick response! I checked out the PR and tested it out in our project, and `constrained` is exactly what I needed. 

I assume the migration path for `both` schemas with multiple predicates is to compose `constrained` schemas?

e.g.

```clojure
(sc/both sc/Int
         (sc/pred pos? 'positive)
         (sc/pred even? 'even))
```

becomes

```clojure
(-> sc/Int
    (sc/constrained pos? 'positive)
    (sc/constrained even? 'even))
```

**Comment by w01fe** [comment]
Released in 1.0.2.  

The above will work just fine (We considered allowing multiple predicates but the syntax didn't seem worth the complexity).  By the way, you should be able to omit the labels for named functions and schema will extract them for you as of 1.0 (e.g. https://github.com/Prismatic/schema/commit/7bb03a9b41eff2ba6955c252f230fcae06b9ed41#diff-b070b5d4dbbb13e87478439b55c05c1fR307)  

Of course, you could also choose to compose the predicates into a single predicate, but I don't see a real advantage there.
